src/BusinessLayer/TelemetryReporting.cpp

```diff
@@ -219,50 +469,83 @@ unsigned int TelemetryReporting::frameData(const unsigned char* dataToEncode,
 {
-   unsigned int lengthOfEncodedData = length + 1;
-   const unsigned char* end = dataToEncode + length;
-   unsigned char* code_ptr = encodedData++;
-   unsigned char code = 0x01;
-
-   while (dataToEncode < end)
-   {
-      if (*dataToEncode == 0)
-      {
-         FINISH_BLOCK(code);
-      }
-      else
-      {
-         *encodedData++ = *dataToEncode;
-         code++;
-         if (code == 0xFF)
-         {
+    unsigned int lengthOfEncodedData = length + 1;
+    const unsigned char* end = dataToEncode + length;
+    unsigned char* code_ptr = encodedData++;
+    unsigned char code = 0x01;
+
+    while (dataToEncode < end)
+    {
+        if (*dataToEncode == 0)
+        {
             FINISH_BLOCK(code);
-            lengthOfEncodedData++;
-         }
-      }
-      dataToEncode++;
-   }
-   FINISH_BLOCK(code);
-   return lengthOfEncodedData;
+        }
+        else
+        {
+            *encodedData++ = *dataToEncode;
+            code++;
+            if (code == 0xFF)
+            {
+                FINISH_BLOCK(code);
+                lengthOfEncodedData++;
+            }
+        }
+        dataToEncode++;
+    }
+    FINISH_BLOCK(code);
+    return lengthOfEncodedData;
 }
 #undef FINISH_BLOCK
 
 void TelemetryReporting::addChecksum(unsigned char* data, unsigned int length)
 {
-   unsigned short crc16 = CrcCalculator::calculateCrc16(data, length);
-   data[length] = static_cast<unsigned char>(0xFF & crc16);
-   data[length + 1] = static_cast<unsigned char>(0xFF & (crc16 >> 8));
+    unsigned short crc16 = CrcCalculator::calculateCrc16(data, length);
+    data[length] = static_cast<unsigned char>(0xFF & crc16);
+    data[length + 1] = static_cast<unsigned char>(0xFF & (crc16 >> 8));
 }
 
 void TelemetryReporting::writeFloatIntoArray(unsigned char* data, int index, const float& value)
 {
-   FloatDataUnion floatDataUnion;
-   floatDataUnion.floatData = value;
-   data[index++] = floatDataUnion.charData[0];
-   data[index++] = floatDataUnion.charData[1];
-   data[index++] = floatDataUnion.charData[2];
-   data[index] = floatDataUnion.charData[3];
+    DataUnion dataUnion;
+    dataUnion.floatData = value;
+    data[index++] = dataUnion.charData[0];
+    data[index++] = dataUnion.charData[1];
+    data[index++] = dataUnion.charData[2];
+    data[index] = dataUnion.charData[3];
+}
+
+void TelemetryReporting::writeShortIntoArray(unsigned char* data, int index, const short& value)
+{
+    DataUnion dataUnion;
+    dataUnion.shortData = value;
+    data[index++] = dataUnion.charData[0];
+    data[index] = dataUnion.charData[1];
+}
+
+void TelemetryReporting::writeUShortIntoArray(unsigned char* data, int index, const unsigned short& value)
+{
+    DataUnion dataUnion;
+    dataUnion.uShortData = value;
+    data[index++] = dataUnion.charData[0];
+    data[index] = dataUnion.charData[1];
+}
+
+void TelemetryReporting::writeBoolsIntoArray(unsigned char* data, int index, const bool values[], int numValues)

```

Comment:
Using `const std::bitset&` as a parameter doesn't compile because the bitset size needs to be determined at compile time. I could template the function, but I don't think it's worth it. Unless you really think otherwise, I'm going to keep it as is.